shared/provision/set-public-name/index.tsx

```diff
@@ -62,13 +71,21 @@ const SetPublicName = (props: Props) => {
         <Kb.Box2 direction="vertical" style={styles.wrapper} gap="xsmall">
           <Kb.NewInput
             autoFocus={true}
+            error={disabled}
             placeholder="Pick a device name"
             onEnterKeyDown={onSubmit}
             onChangeText={setDeviceName}
```

Comment:
we could just filter out illegal characters here, by passing it through a filter before calling `setDeviceName`